### PR TITLE
feat(date-picker): initial design guidance

### PR DESCRIPTION
**Pull request by venkateshr06**
Added initial guidance for datepicker component.
[CDE-2378](https://vmw-jira.broadcom.net/browse/CDE-2378) - Create MD file for DatePicker Component

### CODE REVIEW

File: 1010-datepicker-design-guidance.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-15
+updated: 2024-10-22
+---
+
+# Datepicker Component
+
+[Datepicker](https://clarity.design/documentation/datepicker) component allows users to choose and input a date by manually typing the date into the input field or by using a calendar popover.
```
Comment (kevinbuhmann) [inline 1010-datepicker-design-guidance.md:10]:
```suggestion
The [datepicker](https://clarity.design/documentation/datepicker) component allows users to choose and input a date by manually typing the date into the input field or by using a calendar popover.
```

File: 1010-datepicker-design-guidance.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-15
+updated: 2024-10-22
+---
+
+# Datepicker Component
```
Comment (kevinbuhmann) [inline 1010-datepicker-design-guidance.md:8]:
```suggestion
# Date Picker Design Guidance
```

Please also rename this file to match.

File: 1010-datepicker-design-guidance.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+# Date Picker Design Guidance
+
+The [datepicker](https://clarity.design/documentation/datepicker) component allows users to choose and input a date by manually typing the date into the input field or by using a calendar popover.
+
+## Guidance
+
+- **Must not** use for entering date of birth. Use input component instead.
+- **Must not** use to choose a date that is over 10 years in the future or the past.
```
Comment (jbusta5) [inline 1010-datepicker-design-guidance.md:15]:
```suggestion
- Design teams **must not** use the datepicker to select a date that is more than 10 years in the future or past.
```

File: 1010-datepicker-design-guidance.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- **Must not** use date picker when space is limited.
```
Comment (jbusta5) [inline 1010-datepicker-design-guidance.md:16]:
```suggestion
- Design teams **must not** use the datepicker when space is limited.
```

File: 1010-datepicker-design-guidance.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- **Must** keep the size of the datepicker widget the same, regardless of text field width.
```
Comment (jbusta5) [inline 1010-datepicker-design-guidance.md:17]:
```suggestion
- Design teams **must** keep the size of the datepicker widget consistent, regardless of the text field width.
```

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**